packages/code-du-travail-data/dataset/code_du_travail/fetch.js

```diff
@@ -0,0 +1,129 @@
+const fs = require("fs");
+const path = require("path");
+const serialExec = require("promise-serial-exec");
+const DilaApi = require("@socialgouv/dila-api-client");
+
+const dilaClient = new DilaApi();
+
+const tableMatieres = params =>
+  dilaClient.fetch({
+    path: "consult/code/tableMatieres",
+    method: "POST",
+    params
+  });
+
+const getArticle = (id, tries = 0) =>
+  dilaClient
+    .fetch({
+      path: "consult/getArticle",
+      method: "POST",
+      params: {
+        id
+      }
+    })
+    // try 3 tentatives
+    .catch(e => {
+      console.log("getArticle.catch", e);
+      if (tries < 3) {
+        return getArticle(id, tries + 1);
+      }
+      throw e;
+    });
+
+const JSONLog = data => console.log(JSON.stringify(data, null, 2));
+
+// parties dites anciennes
+const uselessSections = [
+  "LEGISCTA000006101375", // Partie législative ancienne
+  "LEGISCTA000006107985", // Partie réglementaire ancienne - Décrets en Conseil d'Etat
+  "LEGISCTA000006169868" // Partie réglementaire ancienne - Décrets simples
+];
+
+const isUselessSection = section => uselessSections.includes(section.id);
+
+// embed article details into the section
+const embedArticles = async section => ({
+  ...section,
+  articles: await Promise.all(
```

Comment:
why not use pLimit to avoid to many request at the same time or to queued request ?